runtime/src/in_mem_accounts_index.rs

```diff
@@ -1000,57 +985,101 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
-                            disk.try_write(&k, (&slot_list, v.ref_count()))
-                        };
-                        match disk_resize {
-                            Ok(_) => {
-                                // successfully written to disk
-                                flush_entries_updated_on_disk += 1;
-                                break;
+                                                        // consider whether to write to disk for all the items we may evict, whether evicting due to age or random
+                for (is_random, check_for_eviction_and_dirty) in [
+                    (false, &mut evictions_age_possible),
+                    (true, &mut evictions_random),
+                ] {
+                    for (k, v) in check_for_eviction_and_dirty {
+                        let v = v.take().unwrap();
+                        let mut slot_list = None;
+                        if !is_random {
+                            let mut mse = Measure::start("flush_should_evict");
+                            let (evict_for_age, slot_list_temp) = self.should_evict_from_mem(
+                                current_age,
+                                &v,
+                                startup,
+                                true,
+                                exceeds_budget,
+                            );
+                            slot_list = slot_list_temp;
+                            mse.stop();
+                            flush_should_evict_us += mse.as_us();
+                            if evict_for_age {
+                                evictions_age.push(*k);
+                            } else {
+                                // not evicting, so don't write, even if dirty
+                                continue;
                             }
-                            Err(err) => {
-                                // disk needs to resize. This item did not get resized. Resize and try again.
-                                let m = Measure::start("flush_grow");
-                                disk.grow(err);
-                                Self::update_time_stat(&self.stats().flush_grow_us, m);
+                        }
+                        // if we are evicting it, then we need to update disk if we're dirty
+                        if v.clear_dirty() {
+                            // step 1: clear the dirty flag
+                            // step 2: perform the update on disk based on the fields in the entry
+                            // If a parallel operation dirties the item again - even while this flush is occurring,
+                            //  the last thing the writer will do, after updating contents, is set_dirty(true)
+                            //  That prevents dropping an item from cache before disk is updated to latest in mem.
+                            // happens inside of lock on in-mem cache. This is because of deleting items
```

Comment:
Did this comment get moved/split? "happens inside of lock on in-mem cache."

I see it's the same in the diff on the original side too, so wasn't an issue introduced in this PR and can be addressed in a later PR.